dogen/generator.py

```diff
@@ -185,7 +181,34 @@ def _handle_custom_repo_files(self):
         for f in repo_files:
             self.cfg['additional_repos'].append(os.path.splitext(os.path.basename(f))[0])
 
+    def _validate_cfg(self):
+        """
+        Open and parse the YAML configuration file and ensure it matches
+        our Schema for a Dogen configuration.
+        """
+        # Fail early if descriptor file is not found
+        if not os.path.exists(self.descriptor):
+            raise Error("Descriptor file '%s' could not be found. Please make sure you specified correct path." % self.descriptor)
+
+        schema_path = os.path.join(self.pwd, "schema", "kwalify_schema.yaml")
```

Comment:
`self.pwd` is set to `os.path.realpath(os.path.dirname(os.path.realpath(__file__)))` in the `Generator.__init__`method. I think this means that this will always be relative to the generator.py file, wherever that is. `self.dogen.` isn't (yet) defined for `Generator`